src/components/messages/messages.ts

```diff
@@ -2,18 +2,128 @@ import Component from 'core/component';
 
 import { ComponentName } from 'helpers/const';
 
+import { withStore } from 'HOCs/with-store';
+
+import { socket } from 'socket/socket';
+
+import type { Store } from 'store/store';
+
 import './messages.pcss';
 
-export class Messages extends Component {
+type MessagesProps = {
+  userId: string;
+  selectedChatId: string;
+  token: string;
+  messages: Message[];
+  store: Store;
+  onClick: () => void;
+};
+
+export class Messages extends Component<MessagesProps> {
   static componentName = ComponentName.Messages;
 
+  _socket: WebSocket | null;
+
+  constructor(props: MessagesProps) {
+    super(props);
+
+    this._socket = null;
+  }
+
+  componentDidUpdate(oldProps: Partial<any>, newProps: Partial<any>): boolean {
+    if (oldProps.selectedChatId !== newProps.selectedChatId) {
+      if (this._socket !== null) {
```

Comment:
Обычно такие проверки сокращают до 
```suggestion
      if (!this._socket) {
```